autogpt_platform/frontend/src/lib/autogpt-server-api/client.ts

```diff
@@ -1,15 +1,669 @@
 import { SupabaseClient } from "@supabase/supabase-js";
-import { createClient } from "../supabase/client";
-import BaseAutoGPTServerAPI from "./baseClient";
+import {
+  AnalyticsDetails,
+  AnalyticsMetrics,
+  APIKeyCredentials,
+  Block,
+  CredentialsDeleteNeedConfirmationResponse,
+  CredentialsDeleteResponse,
+  CredentialsMetaResponse,
+  GraphExecution,
+  Graph,
+  GraphCreatable,
+  GraphExecuteResponse,
+  GraphMeta,
+  GraphUpdateable,
+  NodeExecutionResult,
+  MyAgentsResponse,
+  OAuth2Credentials,
+  ProfileDetails,
+  User,
+  StoreAgentsResponse,
+  StoreAgentDetails,
+  CreatorsResponse,
+  CreatorDetails,
+  StoreSubmissionsResponse,
+  StoreSubmissionRequest,
+  StoreSubmission,
+  StoreReviewCreate,
+  StoreReview,
+  ScheduleCreatable,
+  Schedule,
+} from "./types";
+import { createBrowserClient } from "@supabase/ssr";
+import getServerSupabase from "../supabase/getServerSupabase";
+
+const isClient = typeof window !== "undefined";
+
+export default class BackendAPI {
+  private baseUrl: string;
+  private wsUrl: string;
+  private webSocket: WebSocket | null = null;
+  private wsConnecting: Promise<void> | null = null;
+  private wsMessageHandlers: Record<string, Set<(data: any) => void>> = {};
+  heartbeatInterval: number | null = null;
+  readonly HEARTBEAT_INTERVAL = 10_0000; // 100 seconds
+  readonly HEARTBEAT_TIMEOUT = 10_000; // 10 seconds
+  heartbeatTimeoutId: number | null = null;
 
-export class AutoGPTServerAPI extends BaseAutoGPTServerAPI {
   constructor(
     baseUrl: string = process.env.NEXT_PUBLIC_AGPT_SERVER_URL ||
       "http://localhost:8006/api",
     wsUrl: string = process.env.NEXT_PUBLIC_AGPT_WS_SERVER_URL ||
       "ws://localhost:8001/ws",
-    supabaseClient: SupabaseClient | null = createClient(),
   ) {
-    super(baseUrl, wsUrl, supabaseClient);
+    this.baseUrl = baseUrl;
+    this.wsUrl = wsUrl;
+  }
+
+  private get supabaseClient(): SupabaseClient | null {
+    return isClient
+      ? createBrowserClient(
+          process.env.NEXT_PUBLIC_SUPABASE_URL!,
+          process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY!,
+        )
+      : getServerSupabase();
+  }
```

Comment:
Reason for getter and not caching:
> Do I need to create a new client for every route?

> Yes! Creating a Supabase client is lightweight.
> - On the server, it basically configures a fetch call. You need to reconfigure the fetch call anew for every request to your server, because you need the cookies from the request.
> - On the client, createBrowserClient already uses a singleton pattern, so you only ever create one instance, no matter how many times you call your createClient function.

[Source](https://supabase.com/docs/guides/auth/server-side/nextjs)